src/analysis/complex/conformal.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import analysis.complex.isometry
+import analysis.normed_space.conformal_linear_map
+
+/-!
+# Conformal maps between complex vector spaces
+
+We prove the sufficient and necessary conditions for a real-linear map between complex vector spaces
+to be conformal.
+
+## Main results
+
+* `is_conformal_map_complex_linear`: a nonzero complex linear map into an arbitrary complex
+                                     normed space is conformal.
+* `is_conformal_map_complex_linear_conj`: the composition of a nonzero complex linear map with
+                                          `conj` is complex linear.
+* `is_complex_or_conj_complex_linear_iff_is_conformal_map`: a real linear map between the complex
+                                                            plane is conformal iff it's complex
+                                                            linear or the composition of
+                                                            some complex linear map and `conj`.
+
+## Warning
+
+Antiholomorphic functions such as the complex conjugate are considered as conformal functions in
+this file.
+-/
+
+noncomputable theory
+
+open complex continuous_linear_map
+
+lemma is_conformal_map_conj : is_conformal_map (conj_lie : ℂ →L[ℝ] ℂ) :=
+(conj_lie.to_linear_isometry).is_conformal_map
+
+section conformal_into_complex_normed
+
+variables {E : Type*} [normed_group E] [normed_space ℝ E] [normed_space ℂ E]
+  [is_scalar_tower ℝ ℂ E] {z : ℂ} {g : ℂ →L[ℝ] E} {f : ℂ → E}
+
+lemma is_conformal_map_complex_linear
+  {map : ℂ →L[ℂ] E} (nonzero : map ≠ 0) : is_conformal_map (map.restrict_scalars ℝ) :=
+begin
+  have minor₀ : ∀ (x : ℂ), x = x • 1 := λ x, by rw [smul_eq_mul, mul_one],
+  have minor₁ : ∥map 1∥ ≠ 0,
+  { contrapose! nonzero with w,
+    ext1,
+    rw [continuous_linear_map.zero_apply],
+    exact norm_eq_zero.mp w, },
+  refine ⟨∥map 1∥, minor₁, _⟩,
```

Comment:
```suggestion
  refine ⟨∥map 1∥, minor₁, ⟨∥map 1∥⁻¹ • map, _⟩, _⟩,
```
I would tend to do this upfront, so that `li'` and `li` never receive an explicit definition, and the two `refine`s are combined into one.